xarray/namedarray/core.py

```diff
@@ -446,13 +473,104 @@ def copy(
         """
         return self._copy(deep=deep, data=data)
 
+    def reduce(
+        self,
+        func: Callable[..., Any],
+        dim: Dims = None,
+        axis: int | Sequence[int] | None = None,
```

Comment:
Is NamedArray big fans of axis inputs? I thought this was a mistake in xarray.

Reply:
Ah I forgot to raise this as an issue. It's definitely a mistake for Dataset. DataArray and Variable do have ordered dimensions so in principle it's OK but discouraged.